From f86694990b2e151e81f3fdf822f3218c8088a628 Mon Sep 17 00:00:00 2001
From: Timo Mueller <timo.mueller@bmw-carit.de>
Date: Wed, 1 Apr 2015 15:23:10 +0200
Subject: [PATCH] Remove inline-block from wrapper element

---
 dist/typeahead.jquery.js | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/dist/typeahead.jquery.js b/dist/typeahead.jquery.js
index 2e4977d..5513a10 100644
--- a/dist/typeahead.jquery.js
+++ b/dist/typeahead.jquery.js
@@ -139,8 +139,7 @@
         "use strict";
         var css = {
             wrapper: {
-                position: "relative",
-                display: "inline-block"
+                position: "relative"
             },
             hint: {
                 position: "absolute",
@@ -1181,4 +1180,4 @@
             return this;
         };
     })();
-})(window.jQuery);
\ No newline at end of file
+})(window.jQuery);
-- 
2.1.0

